Library/Homebrew/cmd/test-bot.rb

```diff
@@ -981,4 +983,35 @@ def test_bot
 
     Homebrew.failed = any_errors
   end
+
+  def sanitize_output_for_xml(output)
+    unless output.empty?
+      # Remove invalid XML CData characters from step output.
+      if ruby_has_encoding?
+        # This is the regex for valid XML chars, but only works in Ruby 2.0+
+        # /[\x09\x0A\x0D\x20-\uD7FF\uE000-\uFFFD\u{10000}-\u{10FFFF}]/
+        # For 1.9 compatibility, use the inverse of that, which stays under \u10000
```

Comment:
Yes, sorry if my comment here isn't clear: I mean that while 1.9 introduced encodings and `\u` escapes, only 2.0 introduced the `\u{...}` escapes that allow more than 4 digits for representing values U+10000 and above. So the original pattern `/[\x09\x0A\x0D\x20-\uD7FF\uE000-\uFFFD\u{10000}-\u{10FFFF}]/` blows up in 1.9. I inverted the pattern so it could be written with just the `\uXXXX` escapes and not `\u{...}`, so it would work in 1.9, too.

Maybe it could be worded better?